src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +153,58 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @RequestMapping(value = "json/experiments/check/zip",
+            method = RequestMethod.GET,
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+        for (var accession : accessions) {
+            try {
+                experiments.add(experimentTrader.getPublicExperiment(accession));
+            } catch (Exception e) {
+                LOGGER.debug("Invalid experiment accession: {}", accession);
+            }
+        }
+        List<ExperimentFileType> fileTypeCheckList = new ArrayList<>(Arrays.asList(
```

Comment:
Again, `ImmutableList.of(...)` is your friend here. Also, at the expense of seeming a bit picky, these are not file types, but just files.

Reply:
I reformat the code but I think it is file type so I did not change the variable name.